tortoise/contrib/test/__init__.py

```diff
@@ -345,11 +346,15 @@ def __init__(self, connection) -> None:
     async def __aenter__(self):
         current_transaction = current_transaction_map[self.connection_name]
         self.token = current_transaction.set(self.connection)
+        if hasattr(self.connection, "_parent"):
+            self.connection._connection = await self.connection._parent._pool.acquire()
```

Comment:
Could you please explain as to why you are doing this? Is aquiring connections not supposed to be managed safely by the transaction wrapper?

Reply:
the problem is you need to acquire a connection from the pool to interact with the database, to keep a transaction intact it needs to be using the same connection

the code for making starting a connection is sync and not async so this is the first async context, also by using the enter and exit functions i can make 100% we always return the connection to the pool no matter what instead of leaking upon error

Reply:
Interesting... I will have to play with it myself to completely understand it.